🌱 feat(e2e): Dump redacted Secret YAMLs for debuggability

shivansh-gohem opened the pull request:
Hi maintainers,

This PR is ready for review. It addresses issue #3184 by adding a diagnostic step to the e2e test workflow that dumps redacted Secret YAMLs. This should help improve debuggability as discussed in the parent epic.

I have successfully run the full `./run-test.sh` suite locally, and all tests passed.

Please let me know if any changes are needed. Looking forward to your feedback!

Thanks.

MikeSpreitzer commented on .github/workflows/pr-test-e2e.yml:
```diff
@@ -183,6 +182,32 @@ jobs:
+      - name: Dump redacted Secret YAMLs
+        if: always()
+        run: |
+          echo "--- Dumping redacted Secret YAMLs from context kind-kubeflex ---"
```
IMO this header is not necessary, the workflow structure already distinguishes this output from other. Not a blocker.

shivansh-gohem commented:
Hi @MikeSpreitzer, thanks again for the guidance! I've updated the script to handle stringData and use the more succinct jq command as you suggested. I also removed the header. This should be ready now.

MikeSpreitzer commented on .github/workflows/pr-test-e2e.yml:
```diff
@@ -317,3 +337,24 @@ jobs:
+              # Use the more succinct `map_values` function to redact all keys, then format as YAML.
+              echo "$COMBINED_DATA" | jq '{"data": map_values("<redacted>")}' | yq -P
+            fi
+          done
```
Not loving the alert here.
<img width="197" height="57" alt="image" src="https://github.com/user-attachments/assets/abbcd934-9386-4159-9769-48fb47eae480" />

shivansh-gohem commented:
Hi @MikeSpreitzer, thanks again for the clear feedback. I've updated the script to handle data and stringData separately as you pointed out. I believe this should be the final version. Please let me know if there's anything else.

shivansh-gohem commented:
Hi @MikeSpreitzer, I've pushed a fix for the PR title check (added the 🌱 emoji).

The Run integration tests / Test job is still failing with that flaky "address already in use" error. Since I don't have permissions to restart it, could you please trigger a re-run for that job?

Thanks!
